pandas/core/frame.py

```diff
@@ -3921,16 +3921,19 @@ def _set_value(
             Sets whether or not index/col interpreted as indexers
         """
         try:
+            # setitem will do validation that may raise TypeError,
+            #  ValueError, or LossySetitemError
             if takeable:
-                series = self._ixs(col, axis=1)
-                loc = index
+                icol = col
+                iindex = index
             else:
-                series = self._get_item_cache(col)
-                loc = self.index.get_loc(index)
-
-            # setitem_inplace will do validation that may raise TypeError,
-            #  ValueError, or LossySetitemError
-            series._mgr.setitem_inplace(loc, value)
+                icol = self.columns.get_loc(col)
+                iindex = self.index.get_loc(index)
+            # error: Argument 2 to "column_setitem" of "BlockManager" has
```

Comment:
might be unnecessary with a `cast` in the takeable case

Reply:
I tried that before adding the ignore, but I don't really understand why it didn't work